aicsimageio/readers/glob_reader.py

```diff
@@ -0,0 +1,541 @@
+            darr = darr.reshape(reshape_sizes)
+            darr = darr.transpose(axes_order)
+            d_data = darr.reshape(tuple(chunk_sizes.values()))
+            dims = list(expanded_chunk_sizes.keys())
+
+        # Assign dims and coords to construct xarray
+        channel_names = self._get_channel_names_for_scene(dims)
+
+        coords = self._get_coords(
+            dims, d_data.shape, self.current_scene_index, channel_names
+        )
+
+        # Try accepted processed metadata
+        try:
+            attrs = {
+                constants.METADATA_UNPROCESSED: tiff_tags,
+                constants.METADATA_PROCESSED: tiff_tags[
+                    TIFF_IMAGE_DESCRIPTION_TAG_INDEX
+                ],
+            }
+        except KeyError:
+            attrs = {constants.METADATA_UNPROCESSED: tiff_tags}
+
+        x_data = xr.DataArray(d_data, dims=dims, coords=coords, attrs=attrs)
+
+        x_data = x_data.transpose(*self._dim_order)
+
+        return x_data
+
+    def _read_immediate(self) -> xr.DataArray:
+        # Set up scene specific information
+        scene_files = self._all_files.loc[
+            self._all_files[self.scene_glob_character] == self.current_scene_index
+        ]
+        scene_files = scene_files.drop(self.scene_glob_character, axis=1)
+        scene_nunique = scene_files.nunique()
+
+        tiff_tags = self._get_tiff_tags(TiffFile(scene_files.filename.iloc[0]))
+
+        chunk_sizes = self._get_chunk_sizes(scene_nunique)
+
+        unpack_sizes = OrderedDict(
+            [(d, s) for d, s in scene_nunique.iteritems() if d in chunk_sizes.keys()]
+        )
+
+        reshape_sizes = tuple(unpack_sizes.values()) + tuple(
+            self._single_file_sizes.values()
+        )
+
+        axes_order = self._get_axes_order(chunk_sizes, unpack_sizes)
+        # Assemble array
+        arr = imread(scene_files.filename.tolist(), level=0)
+        arr = arr.reshape(reshape_sizes)
+        arr = arr.transpose(axes_order)
+        arr = arr.reshape(tuple(chunk_sizes.values()))
+
+        # Assign dims and coords to construct xarray
+        dims = scene_files.columns.drop("filename").values.tolist()
+        file_dims = [x for x in self._single_file_dims if x not in dims]
+        dims += file_dims
+
+        channel_names = self._get_channel_names_for_scene(dims)
+
+        coords = self._get_coords(
+            dims, arr.shape, self.current_scene_index, channel_names
+        )
+
+        # Try accepted processed metadata
+        try:
+            attrs = {
+                constants.METADATA_UNPROCESSED: tiff_tags,
+                constants.METADATA_PROCESSED: tiff_tags[
+                    TIFF_IMAGE_DESCRIPTION_TAG_INDEX
+                ],
+            }
+        except KeyError:
+            attrs = {constants.METADATA_UNPROCESSED: tiff_tags}
+        x_data = xr.DataArray(
+            arr,
+            dims=dims,
+            coords=coords,
```

Comment:
attrs missing here

Reply:
Bump, `attrs` is where the metadata should go, I know we haven't decided what to do with metadata yet but I think as a general solution for now we should just have it be the first files metadata / XML / etree.... Opening every file and reading metadata may be expensive and so would rather just have the first than nothing.